components/Shimmer/src/Shimmer/Shimmer.Properties.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace CommunityToolkit.Labs.WinUI;
+public partial class Shimmer : Control
+{
+    public static readonly DependencyProperty DurationProperty = DependencyProperty.Register(
+        nameof(Duration), typeof(TimeSpan), typeof(Shimmer), new PropertyMetadata(TimeSpan.FromMilliseconds(1600), (s, e) =>
+        {
+            var self = (Shimmer)s;
+            if (self.IsActive)
+            {
+                self.TryStartAnimation();
```

Comment:
Do we need to ensure we stop the animation first? (as doesn't this method no-op if it's already running?)

Could we just update the `_gradient*PointAnimation.Duration` properties, since we have a reference now, would that work?

Reply:
Stopping the animation now first - wrt duration, you want to set those to 0 instead?

Reply:
Sorry should have made a separate comment. Let me do that...

Reply:
Well, I think there's two approaches here when the `Duration` property changes, we either just stop/restart the animation to update everything.

Otherwise, I was wondering if we could just modify the duration of in the inner animations:

```cs
_gradientStartPointAnimation.Duration = e.NewValue;
_gradientEndPointAnimation.Duration = e.NewValue;
```

But not sure if we'd have to stop and restart the composition animation anyway?

```cs
        _shimmerMaskGradient!.StopAnimation(nameof(CompositionLinearGradientBrush.StartPoint));
        _shimmerMaskGradient.StopAnimation(nameof(CompositionLinearGradientBrush.EndPoint));
        _shimmerMaskGradient!.StartAnimation(nameof(CompositionLinearGradientBrush.StartPoint), _gradientStartPointAnimation);
        _shimmerMaskGradient.StartAnimation(nameof(CompositionLinearGradientBrush.EndPoint), _gradientEndPointAnimation);
```

Which means it may just be easier/simpler to call Stop and then start. Which also means we could abstract the callback here and re-use as a method for both the `IsActive` property changed and the `Duration` one, maybe that's the simplest?

Reply:
Moved the start/stop to a single method 👍